File: backend/app/schemas/changelog.py
"""Response schemas for the changelog endpoints."""

import datetime

from pydantic import BaseModel


class ChangelogStatusResponse(BaseModel):
    """Response shape for both GET /changelog/status and POST /changelog/mark-seen.

    Attributes:
        last_seen_changelog_at: UTC timestamp of when the user last dismissed
            the changelog, or None if they have never done so.
    """

    last_seen_changelog_at: datetime.datetime | None

File: frontend/src/api/changelog.ts
import apiClient from "./client";

export interface ChangelogStatus {
  last_seen_changelog_at: string | null;
}

export async function fetchChangelogStatus(): Promise<ChangelogStatus> {
  const res = await apiClient.get<ChangelogStatus>("/api/changelog/status");
  return res.data;
}

export async function markChangelogSeen(): Promise<ChangelogStatus> {
  const res = await apiClient.post<ChangelogStatus>("/api/changelog/mark-seen");
  return res.data;
}
